harness/determined/pytorch/_lr_scheduler.py

```diff
@@ -20,15 +28,12 @@ class StepMode(enum.Enum):
         STEP_EVERY_BATCH = 2
         MANUAL_STEP = 3
 
-    def __init__(self, scheduler: torch.optim.lr_scheduler._LRScheduler, step_mode: StepMode):
-        """Wrapper for a PyTorch LRScheduler.
-
-        Usage of this wrapper is required to properly schedule the optimizer's learning rate.
-
-        This wrapper fulfills two main functions:
-            1. Save and restore the learning rate when a trial is paused, preempted, etc.
-            2. Step the learning rate scheduler at the configured frequency
-               (e.g., every batch or every epoch).
+    def __init__(
+        self,
+        scheduler: torch.optim.lr_scheduler._LRScheduler,
+        step_mode: StepMode = StepMode.MANUAL_STEP,
```

Comment:
blocking: we originally had a default StepMode, but users ran into issues with that case, and we made it a required parameter.  I personally think it should stay a required parameter.  Perhaps as an alternative to having a default mode, we could introduce another classes like `det.pytorch.AutoStepLRScheduler()` that users are not allowed to call `step()` on at all.

But in this PR, leaving the user-facing API unchanged seems pretty reasonable to me.